src/commands/validate-address.ts

```diff
@@ -0,0 +1,84 @@
+import { Command, flags } from '@oclif/command';
+import { c32addressDecode, versions as NetworkVersions } from 'c32check';
+
+type NetworkString = 'mocknet' | 'mainnet' | 'testnet';
+
+export class ValidateAddress extends Command {
+	static description = `Validates whether the input is a valid STX address for the provided network.
+
+  Example usage:
+
+  \`\`\`
+  npx stx-bulk-transfer validate-address SP000000000000000000002Q6VF78 -n mainnet
+  \`\`\`
+  `;
+	static strict = true;
+
+	static flags = {
+		help: flags.help({ char: 'h' }),
+		network: flags.string({
+			char: 'n',
+			description: 'Which network to check for',
+			options: ['mocknet', 'testnet', 'mainnet'],
+			default: 'mainnet',
+		}),
+		verbose: flags.boolean({
+			required: false,
+			char: 'v',
+			default: false,
+			description: 'Print error information for invalid addresses',
+		})
+	};
+
+	static args = [
+		{
+			name: 'address',
+			description: `The address to validate`,
+		},
+	];
+
+	getNetworkVersion() {
+		const { flags } = this.parse(ValidateAddress);
+
+		const networks = {
+			mainnet: NetworkVersions.mainnet,
+			testnet: NetworkVersions.testnet,
+			mocknet: NetworkVersions.testnet,
+		};
+
+		return networks[flags.network as NetworkString];
+	}
+
+	async run() {
+		const { argv, flags } = this.parse(ValidateAddress);
+		const [address] = argv;
+
+		if (!address) {
+			throw new Error('No address specified, try --help');
+		}
+
+		const networkVersion = this.getNetworkVersion();
+		if (!networkVersion) {
+			throw new Error('Invalid network');
+		}
+
+		try {
+			const [version] = c32addressDecode(address);
+			if (version === networkVersion.p2pkh || version === networkVersion.p2sh) {
+				this.log('1');
+			}
+			else {
+				this.log('0');
+				if (flags.verbose) {
+					this.log(`Valid address but incorrect network version (address version: ${version}, expected: ${networkVersion.p2pkh} or ${networkVersion.p2sh})`);
+				}
+			}
+		}
+		catch (error) {
+			this.log('0');
+			if (flags.verbose) {
+				this.log(error as any);
+			}
+		}
```

Comment:
Should this new command also specify a bad (non-zero) process exit code for invalid addresses? That would allow it to work as expected with other CLI tooling, like 
`stx-bulk-transfer validate-address SM1D5WZDAR17J8S4GSYTB8PS9V5WKAZNZA0DZ76BZ && run-send-command`

Reply:
Great idea! Why did I not think of it.